[AOTI] Flag to include aoti sources when building lite interpreter

manuelcandales opened the pull request:
Summary:
Added USE_LITE_AOTI cmake flag, which is turned OFF by default.
When it is turned on, the AOTI sources  (inductor_core_resources) are included when building lite interpreter

Test Plan:
```
ANDROID_ABI=arm64-v8a ./scripts/build_android.sh -DUSE_LITE_AOTI=ON
```

Differential Revision: D57394078

malfet commented on caffe2/CMakeLists.txt:
```diff
@@ -452,11 +452,17 @@ if(USE_LITE_INTERPRETER_PROFILER)
   append_filelist("libtorch_edge_profiler_sources " LITE_PROFILER_SRCS)
 endif()
 
+list(APPEND LITE_AOTI_SRCS "")
+if(USE_LITE_AOTI)
+  append_filelist("inductor_core_resources" LITE_AOTI_SRCS)
+endif()
```
Easiest solution is usually the best
```suggestion
```

malfet commented on caffe2/CMakeLists.txt:
```diff
@@ -452,11 +452,17 @@ if(USE_LITE_INTERPRETER_PROFILER)
   append_filelist("libtorch_lite_cmake_sources" LIBTORCH_CMAKE_SRCS)
   list(APPEND LIBTORCH_CMAKE_SRCS ${LITE_EAGER_SYMOBLICATION_SRCS})
   list(APPEND LIBTORCH_CMAKE_SRCS ${LITE_PROFILER_SRCS})
+  list(APPEND LIBTORCH_CMAKE_SRCS ${LITE_AOTI_SRCS})
```
Why not do something like that?
```suggestion
  if(USE_LITE_AOTI)
    append_filelist("inductor_core_resources" LIBTORCH_CMAKE_SRCS)
  endif()

    list(APPEND LIBTORCH_CMAKE_SRCS ${LITE_AOTI_SRCS})
```

malfet commented on caffe2/CMakeLists.txt:
```diff
@@ -457,6 +457,9 @@ if(BUILD_LITE_INTERPRETER)
   list(APPEND LIBTORCH_CMAKE_SRCS ${LITE_EAGER_SYMOBLICATION_SRCS})
   list(APPEND LIBTORCH_CMAKE_SRCS ${LITE_PROFILER_SRCS})
+  if(USE_LITE_AOTI)
+    append_filelist("inductor_core_resources" LITE_AOTI_SRCS)
```
```suggestion
    append_filelist("inductor_core_resources" LIBTORCH_CMAKE_SRCS)
```

malfet commented:
Also, wouldn't `USE_LITE_AOTI=ON ANDROID_ABI=arm64-v8a ./scripts/build_android.sh` would also work?

manuelcandales committed:
[AOTI] Flag to include aoti sources when building lite interpreter (#126572)

Summary:

Added USE_LITE_AOTI cmake flag, which is turned OFF by default.
When it is turned ON, the AOTI sources (inductor_core_resources) are included when building lite interpreter

Test Plan:
```
ANDROID_ABI=arm64-v8a ./scripts/build_android.sh -DUSE_LITE_AOTI=ON
```

Differential Revision: D57394078
